tools/git-version-gen

```diff
@@ -10,7 +10,7 @@ if [ $# -ne 1 ]; then
 fi
 
 generate() {
-	if [ -e $BASE/../.git ]; then
+	if [ -e $BASE/../.git ] && [ git describe 2>/dev/null ]; then
```

Comment:
It's rather weird to use `[ ]` for an exit code test, and it's not a correct usage of `test`. Also, this repeats the `git describe`.

This may be the reason for all semaphore tests failing:

    configure: error: Please build from a .git checkout or .tar.xz tarball

Could you please rather rewrite it like this?
```sh
    if desc=$(git -C "$BASE/.." describe); then
        echo "$desc" | sed 's/-[0-9]\+-g.*/.x/'
    else
        cat "$1"
fi
```

This would also address the missing quoting and wrong indentation as a side effect. :-)

Reply:
This is a very good proposal. :+1:  I will commit the adjustment.

Hm... I should improve my shell knowledge :-)